docs/docs/rest-api/public/api/v2/types/deploymentEventsInfo.raml

```diff
@@ -0,0 +1,42 @@
+#%RAML 1.0 Library
+uses:
+  strings: stringTypes.raml
+  deploymentAction: deploymentAction.raml
+  deploymentPlan: deploymentPlan.raml
+
+types:
+
+  # Annotate the deployment action with more information required for the event
+  DeploymentActionInfo:
+    type: deploymentAction.DeploymentAction
```

Comment:
Why inherit from `DeploymentAction`?

Reply:
The `DeploymentActionInfo` provides additional meta-information required by the event version of `DeploymentAction`. You can find the missing fields in the [`DeploymentStepInfoWrites` formatter](https://github.com/mesosphere/marathon/blob/master/src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala#L188).

I Quote:

```scala
  implicit lazy val DeploymentStepInfoWrites: Writes[DeploymentStepInfo] = Writes { info =>
    def currentAction(action: DeploymentAction): JsObject = Json.obj (
      "action" -> DeploymentAction.actionName(action),
      actionInstanceOn(action.runSpec) -> action.runSpec.id,
      "readinessCheckResults" -> info.readinessChecksByApp(action.runSpec.id)
    )
    Json.obj(
      "id" -> info.plan.id,
      "version" -> info.plan.version,
      "affectedApps" -> info.plan.affectedAppIds,
      "affectedPods" -> info.plan.affectedPodIds,
      "steps" -> info.plan.steps,
      "currentActions" -> info.step.actions.map(currentAction),
      "currentStep" -> info.stepIndex,
      "totalSteps" -> info.plan.steps.size
    )
  }

```

Reply:
Hm, for me an inheritance is an `is a` relation. So it'd say `DeploymentActionInfo is a DeploymentAction`. However, I don't think this is a proper relation. I'd rather not encode this relation in the RAML.